src/yunohost/migrations.py

```diff
@@ -0,0 +1,249 @@
+
+    Migration framework for our data modifications.
+"""
+
+import os
+import re
+import json
+import errno
+from importlib import import_module
+from moulinette.utils.log import getActionLogger
+from moulinette.core import MoulinetteError
+
+
+MIGRATIONS_STATE_PATH = "/etc/yunohost/migrations_state.json"
+
+logger = getActionLogger('yunohost.migrations')
+
+
+def migrations_list():
+    """
+    List existing migrations
+    """
+
+    migrations = {"migrations": []}
+
+    for migration in _get_migrations_list():
+        migrations["migrations"].append({
+            "number": int(migration.split("_", 1)[0]),
+            "name": migration.split("_", 1)[1],
+            "file_name": migration,
+        })
+
+    return migrations
+
+
+# TODO need a "fake" option and also a migration number as possible argument (same than in django)
+def migrations_migrate(target=None, fake=False):
+    """
+    Perform migrations
+    """
+
+    # state is a datastructure of that represent the latest runed migration
+    # it has this form:
+    # {
+    #     "last_runed_migration": {
+    #             "number": "00xx",
+    #             "name": "some name",
+    #         }
+    # }
+    state = migrations_state()
+
+    last_runed_migration_number = state["last_runed_migration"]["number"] if state["last_runed_migration"] else 0
+
+    migrations = []
+
+    # loading all migrations
+    for migration in migrations_list()["migrations"]:
+        logger.debug("Loading migration {number} {name}...".format(
+            number=migration["number"],
+            name=migration["name"],
+        ))
+
+        try:
+            module = import_module("yunohost.data_migrations.{file_name}".format(**migration))
+        except Exception:
+            import traceback
+            traceback.print_exc()
+
+            raise MoulinetteError(errno.EINVAL, "WARNING: failed to load migration {number} {name}".format(
+                number=migration["number"],
+                name=migration["name"],
+            ))
+            break
+
+        migrations.append({
+            "number": migration["number"],
+            "name": migration["name"],
+            # this is python builtin method to import a module using a name, we use that to import the migration
+            # has a python object so we'll be able to run it in the next loop
+            "module": module,
+        })
+
+    migrations = sorted(migrations, key=lambda x: x["number"])
+
+    if not migrations:
+        logger.info("No migrations to run.")
+        return
+
+    all_migration_numbers = [x["number"] for x in migrations]
+
+    if target is None:
+        target = migrations[-1]["number"]
+
+    # validate input, target must be "0" or a valid number
+    elif target != 0 and target not in all_migration_numbers:
+        raise MoulinetteError(errno.EINVAL, "Invalide number for target argument, available migrations numbers are 0 or {}".format(", ".join(map(str, all_migration_numbers))))
```

Comment:
I think that "number" would be a more obvious error message for the user, telling they that they need to uses ... a number.